### PR TITLE
Support gather verb

### PR DESCRIPTION
**Pull request by rudigiesler**

### CODE REVIEW

File: vumi_twilio_api/twiml_parser.py [hunk 1]
```diff
@@ -42,12 +43,95 @@ def from_xml(cls, xml):
...
+            try:
+                numDigits = int(numDigits)
+            except ValueError:
+                raise TwiMLParseError(
+                    "Invalid value %r for numDigits parameter. "
+                    "Must be an integer." % numDigits)
+            if numDigits < 1:
+                raise TwiMLParseError(
+                    "Invalid value %r for numDigits parameter. Must be >=1"
+                    % numDigits)
```
Comment (hodgestar) [inline vumi_twilio_api/twiml_parser.py]: Maybe it's worthwhile factoring out a simple integer checking function?

**Comment by rudigiesler** [comment]
Ready for review.

File: vumi_twilio_api/tests/test_twiml_parser.py [hunk 2]
```diff
@@ -115,3 +126,120 @@ def test_hangup_from_xml(self):
...
+
+    def test_gather_from_xml_valid_sub_verbs(self):
+        """Should have valid sub verbs parsed"""
+        root = ET.Element("Gather")
+        ET.SubElement(root, "Play")
+        gather = Gather.from_xml(root, 'test_url')
+        self.assertEqual(gather.nouns[0].name, 'Play')
+
+    def test_gather_from_xml_invalid_sub_verbs(self):
+        """Shoudl raise an exception for invalid sub verbs"""
```
Comment (hodgestar) [inline vumi_twilio_api/tests/test_twiml_parser.py]: `s/Shoudl/Should/` :)

**Comment by hodgestar** [comment]
Left one minor comment. Looks good though. There are a lot of TODOs, but I'd like to start trying things out with Rapid before we worry too much about being feature complete.

**Comment by rudigiesler** [comment]
The TODOS are mostly things that I don't think are necessary for Rapid, but they're there for if something with Rapid doesn't work, we know where to put the feature that needs to be added. Also to keep a note of what we need to add in future if we want to make it feature complete.

**Comment by rudigiesler** [comment]
Issues addressed.

**Comment by hodgestar** [comment]
:+1: as soon as Travis is happy.

**Pull request merged**